src/gei/Handlers/MigrateRepoCommandHandler.cs

```diff
@@ -202,17 +205,29 @@ private string ExtractGhesBaseUrl(string ghesApiUrl)
         }
 
         _log.LogInformation($"Downloading archive from {gitArchiveUrl}");
-        var gitArchiveContent = await _httpDownloadService.DownloadToBytes(gitArchiveUrl);
+        var gitArchiveFilePath = _fileSystemProvider.GetTempFileName();
+        await _httpDownloadService.DownloadToFile(gitArchiveUrl, gitArchiveFilePath);
+        using var gitArchiveContent = _fileSystemProvider.OpenRead(gitArchiveFilePath);
 
         _log.LogInformation($"Downloading archive from {metadataArchiveUrl}");
-        var metadataArchiveContent = await _httpDownloadService.DownloadToBytes(metadataArchiveUrl);
+        var metadataArchiveFilePath = _fileSystemProvider.GetTempFileName();
+        await _httpDownloadService.DownloadToFile(metadataArchiveUrl, metadataArchiveFilePath);
+        using var metadataArchiveContent = _fileSystemProvider.OpenRead(metadataArchiveFilePath);
 
-        return _awsApi.HasValue() ?
+        try
+        {
+            return _awsApi.HasValue() ?
```

Comment:
nit: the indentation is a bit off here, both awaits need to be indented one level.

Reply:
Resolved in 7d436e04071bf868990e1d32d6276668c1db1a35 :tada: